subsys/bluetooth/host/smp.c

```diff
@@ -2377,12 +2377,12 @@ static uint8_t legacy_pairing_req(struct bt_smp *smp)
 		return 0;
 	}
 
+	atomic_set_bit(&smp->allowed_cmds, BT_SMP_CMD_PAIRING_CONFIRM);
```

Comment:
if this race is the case shouldn't all state be updated before send? and not just allowed commands?

eg here legacy_request_tk() also modifies smp state (but this is the case in other places too)

Reply:
I didn't actually see these race-conditions come into play, but in all other cases allowed commands was set before sending the packet. So this was my first assumption.

I can have a look into if this is a problem for legacy-request-ltk, but will do that at a later time.

Reply:
@sjanc Looks like legacy_pairing_req is a continuation of the pairing procedure and processing this after the packet has been received is probably fine.
However there was another issue with legacy pairing using a fixed passkey, so I have moved the processing of legacy_request_tk as a fix for that.